command/ca/revoke.go

```diff
@@ -405,11 +407,22 @@ func (f *revokeFlow) Revoke(ctx *cli.Context, serial, token string) error {
 	if len(token) == 0 {
 		certFile, keyFile := ctx.String("cert"), ctx.String("key")
 
-		// If there is no token then we must be doing a Revoke over mTLS.
-		var cert tls.Certificate
-		cert, err = tls.LoadX509KeyPair(certFile, keyFile)
+		certPEMBytes, err := ioutil.ReadFile(certFile)
 		if err != nil {
-			return errors.Wrap(err, "error loading certificates")
+			return errors.Wrap(err, "error reading cert")
+		}
+		key, err := pemutil.Read(keyFile)
+		if err != nil {
+			return errors.Wrap(err, "error parsing key")
+		}
+		keyBlock, err := pemutil.Serialize(key)
+		if err != nil {
+			return errors.Wrap(err, "error serializing key")
+		}
+
+		cert, err := tls.X509KeyPair(certPEMBytes, pem.EncodeToMemory(keyBlock))
```

Comment:
You can build the `*tls.Certificate` yourself, with the `x509.Certtificate` and private key. That is basically what `tls.X509KeyPair` does, but it adds some extra validations making sure the public key in the cert matches the private key.

So I guess it's ok like that.